src/main/java/delete_post/interface_adapters/DeletePostPresenter.java

```diff
@@ -2,20 +2,22 @@
 
 import delete_post.use_case.DeletePostOutputBoundary;
 import delete_post.use_case.DeletePostResponseModel;
-import entities.User;
+import delete_post.UI.DeleteErrorView;
 
 public class DeletePostPresenter implements DeletePostOutputBoundary{
 
     private final DeletePostViewModel viewModel;
+    private final Jframe postView;
 
     public DeletePostPresenter(DeletePostViewModel viewModel){
         this.viewModel = viewModel;
     }
     @Override
-    public void prepareFailView(String error){
-        //error popup, no change to posts view or database;
+    public void prepareFailView(){
```

Comment:
I am not entirely sure but I think here, too, you should update the viewModel (like in prepareSuccessView), which then fires a property change and then the view opens the error view, not the presenter itself. That would be more in keeping with the observer pattern we are using.